service/controller/resource/clusterconfigmap/resource.go

```diff
@@ -22,6 +22,7 @@ type Config struct {
 	GetClusterConfigFunc     func(obj interface{}) (v1alpha1.ClusterGuestConfig, error)
 	GetClusterObjectMetaFunc func(obj interface{}) (metav1.ObjectMeta, error)
 	GetWorkerCountFunc       func(obj interface{}) (int, error)
+	GetWorkerMaxCPUCoresFunc func(obj interface{}) (int, bool, error)
```

Comment:
This should be explained with some comments as well. Otherwise it is too much magic and unknown construct.

Reply:
Named parameters so they serve as documentation, should improve readability